app-frontend/src/app/components/map/annotateToolbar/annotateToolbar.controller.js

```diff
@@ -0,0 +1,163 @@
+/* globals L */
+const _ = require('lodash');
+
+export default class AnnotateToolbarController {
+    constructor(
+        $log, $scope, $compile,
+        mapService
+    ) {
+        'ngInject';
+        this.$log = $log;
+        this.$scope = $scope;
+        this.$compile = $compile;
+        this.getMap = () => mapService.getMap(this.mapId);
+    }
+
+    $onInit() {
+        this.getMap().then((mapWrapper) => {
+            this.listeners = [
+                mapWrapper.on(L.Draw.Event.CREATED, this.addShape.bind(this))
+            ];
+            this.setDrawHandlers(mapWrapper);
+        });
+        this.labelInputs = [];
+        this.labelOriginalObject = {};
+    }
+
+    $onDestroy() {
+        this.getMap().then((mapWrapper) => {
+            this.listeners.forEach((listener) => {
+                mapWrapper.off(listener);
+            });
+            mapWrapper.deleteLayers('Annotation');
+        });
+    }
+
+    setDrawHandlers(mapWrapper) {
+        this.drawRectangleHandler = new L.Draw.Rectangle(mapWrapper.map, {
+            shapeOptions: {
+                weight: 2,
+                fillOpacity: 0.2
+            }
+        });
+        this.drawPolygonHandler = new L.Draw.Polygon(mapWrapper.map, {
+            allowIntersection: false,
+            shapeOptions: {
+                weight: 2,
+                fillOpacity: 0.2
+            }
+        });
+        this.drawMarkerHandler = new L.Draw.Marker(mapWrapper.map, {
+            icon: L.divIcon({'className': 'annotate-marker'})
+        });
+    }
+
+    toggleDrawing(shapeType) {
```

Comment:
It looks like this should also handle disabling the `shapeType`s that don't match.

If you click the different shape you get something like this:

![image](https://user-images.githubusercontent.com/2442245/28681227-c4b3a894-72c6-11e7-9e11-78a0351964bd.png)